listeners.py

```diff
@@ -222,6 +223,16 @@ def save_artifacts(self):
         if not self.failed_workers:
             return
 
+        def is_socket(path):
+            return stat.S_ISSOCK(os.lstat(path).st_mode)
```

Comment:
`os.lstat()` is used, so a symlink is not followed, so a symlink to a socket will be accepted to copy and we'll get the same error (I guess).

I think we can ignore all the special files. IOW, do the `os.path.isfile(x) or os.path.isdir(x)` check.

Reply:
You are absolutely right. So let's use `os.stat()` instead that follows symbolic links. I have tested it and it works as expected.

Reply:
Ok, if we check `os.path.isfile(x) or os.path.isdir(x)`, we will not need to check for a socket at all. Fixed.